File: carpet.cpp
// 프로그래머스 카펫
#include <string>
#include <vector>
using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    vector<pair<int,int>> p;
    for (int i=1;i<yellow;i++){
        if (yellow%i==0){
            if (i>yellow/i){
                break;
            }
            p.push_back(make_pair(yellow/i+2, i));
        }
    }
    if (p.empty()) p.push_back(make_pair(3,1));
    for (int i=0;i<p.size();i++){
        if (2*p[i].first + 2*p[i].second == brown){
            answer.push_back(p[i].first);
            answer.push_back(p[i].second+2);
        }
    }
    return answer;
}

File: carpet.py
# 카펫
def solution(brown, yellow):
    answer = []
    # brown은 yellow 영역의 위아래, 양옆을 채우고 모서리 4곳을 더한 값
    # yellow 영역의 가로 세로 크기를 알기 위해서 약수를 구함
    for i in range(1, yellow+1):
        if (yellow % i == 0):
            yr = yellow // i
            yc = i
            if ((yr+yc)*2 + 4 == brown):
                answer.append(yr+2)
                answer.append(yc+2)
                break
    return answer
